runtime/vm/ContinuationHelpers.cpp

```diff
@@ -90,28 +90,47 @@ createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 	return result;
 }
 
-void
-synchronizeWithConcurrentGCScan(J9VMThread *currentThread, J9VMContinuation *continuation)
+j9object_t
+synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuationObject, J9VMContinuation *continuation)
 {
+
 	volatile uintptr_t *localAddr = &continuation->state;
-	/* atomically 'or' (not 'set') continuation->state  with currentThread */
-	uintptr_t oldContinuationState = VM_AtomicSupport::bitOr(localAddr, (uintptr_t)currentThread);
+	uintptr_t oldContinuationState = 0;
+	uintptr_t newContinuationState = 0;
```

Comment:
pull these 2 definitions inside the loop, so it's consistent with the pattern in exitConcurrentGCScan